boaviztapi/model/consumption_profile/consumption_profile.py

```diff
@@ -31,57 +56,47 @@ class CPUConsumptionProfileModel(ConsumptionProfileModel):
     _MODEL_PARAM_NAME = ['a', 'b', 'c', 'd']
 
     def __init__(self):
-        self.cpu_manufacturer = Boattribute(
-            status=Status.NONE
-        )
-        self.cpu_model_range = Boattribute(
+        self.workloads = Boattribute(
             status=Status.NONE,
-            default=self.DEFAULT_CPU_MODEL_RANGE
+            default=self.DEFAULT_WORKLOADS,
+            unit="workload_rate:W"
         )
-        self.workloads = Boattribute(
+        self.params = Boattribute(
             status=Status.NONE,
-            default=self.DEFAULT_WORKLOADS
+            default=self._DEFAULT_MODEL_PARAMS
         )
 
     @property
     def list_workloads(self) -> Tuple[List[float], List[float]]:
-        load = [item['load'] for item in self.workloads.value]
-        power = [item['power'] for item in self.workloads.value]
+        load = [item.load for item in self.workloads.value]
+        power = [item.power for item in self.workloads.value]
         return load, power
 
-    def compute_consumption_profile_model(self) -> Dict[str, float]:
-        base_model = self.lookup_consumption_profile()
-
-        if base_model is None:
-            base_model = self._DEFAULT_MODEL_PARAMS
+    def apply_consumption_profile(self, load_percentage: float) -> float:
+        return self.__log_model(load_percentage, self.params.value['a'], self.params.value['b'], self.params.value['c'],
+                                self.params.value['d'])
 
-        if self.workloads is None:
-            return base_model
+    def apply_multiple_workloads(self, time_workload: List[WorkloadTime]) -> float:
+        total = 0
+        for workload in time_workload:
+            total += (workload.time_percentage / 100) * self.apply_consumption_profile(workload.load_percentage)
+        return total
 
-        final_model = self.__compute_model_adaptation(base_model)
-        return final_model
+    def compute_consumption_profile_model(self, cpu_manufacturer: str = None, cpu_model_range: str = None) -> Union[Dict[str, float], None]:
+        model = lookup_consumption_profile(cpu_manufacturer, cpu_model_range)
 
-    def lookup_consumption_profile(self) -> Optional[Dict[str, float]]:
-        sub = _cpu_profile_consumption_df
+        if model is None:
+            self.params.value = self._DEFAULT_MODEL_PARAMS
+            self.params.status = Status.DEFAULT
+            return self._DEFAULT_MODEL_PARAMS
 
-        if self.cpu_manufacturer is not None:
-            tmp = sub[sub['manufacturer'] == self.cpu_manufacturer]
-            if len(tmp) > 0:
-                sub = tmp.copy()
+        if self.workloads.status != Status.NONE:
```

Comment:
I see a lot of `if foo.status == Status.NONE:` in the code, I think we should consider refactoring these later with functions like `foo.is_set()`, `foo.is_default()`, etc. Don't need to change it rn, but we should consider it, it will make the code easier to read.

Reply:
I resolve the comment as it is just a suggestion and we can focus on that later.

Reply:
@da-ekchajzer